crates/svm-abi/encoder/src/types/array.rs

```diff
@@ -0,0 +1,40 @@
+extern crate alloc;
+use alloc::vec::Vec;
+
+use svm_abi_layout::layout;
+
+use svm_sdk::types::PrimitiveMarker;
+use svm_sdk::value::Value;
+
+use crate::Encoder;
+
+impl Encoder for &[&dyn Encoder] {
+    fn encode(&self, w: &mut Vec<u8>) {
+        assert!(self.len() < 255);
+
+        let marker = match self.len() {
+            0 => layout::ARR_0,
+            1 => layout::ARR_1,
+            2 => layout::ARR_2,
+            3 => layout::ARR_3,
+            4 => layout::ARR_4,
+            5 => layout::ARR_5,
+            6 => layout::ARR_6,
+            7..256 => layout::ARR_0_255,
```

Comment:
Does this mean that each array with size > 6 will be considered as size = 255?

Reply:
no, it means, that I'll add a `length` field to the encoding

Reply:
It's still not clear to me, can you elaborate more?

Reply:
you can encode `i64` in many ways according to the value.

if the value fits a single-byte you prepend it with a `i64-one-byte` marker
if the value fits a two-byte you prepend it with a `i64-twt-byte` marker 
and so on....

Reply:
So if the value fits a single-byte, why not just use `i8` instead? What's the reasons for preferring `i64`?

Reply:
Also, it still doesn't answer the question regarding the Array size.

Reply:
> So if the value fits a single-byte, why not just use `i8` instead? What's the reasons for preferring `i64`?

if you have a function that expects an `i64` you will call it multiple times with a different value.
when you call it with a small value the encoding will be different from when you'll call it with bigger values.
In both cases, we'll encode an `i64` type. The exact encoding method will be different

Reply:
> Also, it still doesn't answer the question regarding the Array size.

Similarly to the `i64` example.
You encode Arrays. Different inputs can result in completely different implementations.
For Array's larger than 6 - we have `length` field (a single byte)
For Array's larger than 256 (someday, if relevant) - we might want to have another encoding.